### PR TITLE
Liquefied link tag

### PR DESCRIPTION
**Pull request by alexey-pelykh**
This would allow to use

{% assign path = 'some/variable/path' %}
{% assign ext = 'md' %}
{% link {{path}}.{{ext}} %}
in addition to

{% link some/hardcoded/path.md %}

### CODE REVIEW

**Comment by danpalmer** [comment]
Some feedback on this, I think this could be confusing and might not be the best choice for implementation.

The use-case here (please correct me if I'm wrong) is to enable linking to dynamic files. Currently, `link` can only take a hard-coded relative path, but we want to be able to pass something else in.

This implementation re-uses liquid templating, nested within the process of templating another file, in a way which I think could get confusing and be a maintenance burden in the future.

I'd suggest that instead of templating the data passed to `link`, we should allow looking it up as a variable name in the current context. If we did this, your example could be achieved in a similar way with the following code:

```
{% assign path = 'some/variable/path' %}
{% assign fullpath = path|append: ".md" %}
{% link fullpath %}

{% link some/hardcoded/path.md %}
```

This re-uses existing functionality in the templating system to achieve the string building, rather than introducing another layer of templating, and I think this will be simpler to document and maintain.

I'd be interested to hear what you and the maintainers think of these different solutions, and let me know if you want to do this, otherwise I might open a PR with an implementation.

**Comment by alexey-pelykh** [comment]
This implementation is based on #6261, which itself uses the same approach as `include` tag. Current one is more similar to jekyll-assets tags. Anyhow, writing one line instead of three seems more convenient, even if that requires extra depth of templating.

**Comment by alexey-pelykh** [comment]
@danpalmer updated

**Comment by DirtyF** [comment]
@alexey-pelykh Could you document a use case for this? Thanks.

**Comment by alexey-pelykh** [comment]
@DirtyF sure, to which page that should go?

**Comment by DirtyF** [comment]
@alexey-pelykh https://jekyllrb.com/docs/templates/#links please.

**Comment by alexey-pelykh** [comment]
@DirtyF done

File: lib/jekyll/tags/link.rb [hunk 1]
```diff
@@ -18,14 +18,17 @@ def initialize(tag_name, relative_path, tokens)
       def render(context)
         site = context.registers[:site]
 
+        liquid = site.liquid_renderer.file("(jekyll:link)")
```
Comment (ashmaroli) [inline lib/jekyll/tags/link.rb]: @alexey-pelykh can you please explain what `"(jekyll:link)"` is supposed to mean here? That is beyond my Ruby knowledge..

**Comment by pathawks** [comment]
Is this a duplicate of #6261? Seems like one or the other should be closed.

**Comment by alexey-pelykh** [comment]
@pathawks #6261 is super-simple, while this one is "as it should be"

**Comment by pathawks** [comment]
@alexey-pelykh Are you saying _that_ should be closed in favor of _this_?

**Comment by alexey-pelykh** [comment]
Which-ever implementation suits project vision better! I prefer #6269 over #6261 - check the test cases, this one supports more sophisticated usage

**Comment by pathawks** [comment]
Could somebody from @jekyll/windows kick AppVeyor? It seems to have choked.

Otherwise this LGTM :+1:

**Comment by ashmaroli** [comment]
> Could somebody from @jekyll/windows kick AppVeyor? It seems to have choked.

AppVeyor builds cannot be restarted.. The best solution here is pushing the `merge 'master` branch into..` commit.

**Comment by pathawks** [comment]
Thanks for all the work you put into this. Sorry it took so long to get this merged, but this will be a great feature in Jekyll 4.0 :+1::beers:

@jekyllbot: :shipit: +minor

**Pull request merged**